apps/server/src/modules/room/domain/service/room.service.ts

```diff
@@ -46,7 +51,12 @@ export class RoomService {
 
 	public async updateRoom(room: Room, props: RoomUpdateProps): Promise<void> {
 		this.validateTimeSpan(props, room.id);
-		Object.assign(room, props);
+
+		room.name = props.name;
+		room.color = props.color;
+		// make sure that the dates are not null at runtime
```

Comment:
why could that be an issue?

Reply:
One scenario could be when the API is not properly validated and we get `null` values instead of `undefined` at runtime (TS has no runtime type checking). That would lead to `startDate: null` or `endDate: null` in the database. I would prefer one or the other: `null` values *OR* omitted values. Not both.

Reply:
In the API we are explicitely making sure that no `null` values are passed for the optional properties:
```
@IsOptional()
@NullToUndefined()
// ...
```
But that might not be always the case.